modin/core/execution/ray/common/engine_wrapper.py

```diff
@@ -108,6 +111,14 @@ def put(cls, data, **kwargs):
         ray.ObjectID
             Ray object identifier to get the value by.
         """
+        if isinstance(data, FunctionType):
+            qname = data.__qualname__
+            if "<locals>" not in qname and "<lambda>" not in qname:
```

Comment:
Why check `<locals>` as well?

Reply:
To filter out nested functions. I didn't find a better approach than using __qualname__.

Reply:
> To filter out nested functions.

Why?

Reply:
Because of a new function instance on each call. Most probably, the cached functions, will never be reused, it's just a memory leak.